pkg/authorization/api/types.go

```diff
@@ -434,3 +434,86 @@ type ClusterRoleList struct {
 	// Items is a list of ClusterRoles
 	Items []ClusterRole
 }
+
+// CollaboratorMatcher is an object that can be matched against a subject (user,
+// group, or service account) to determine if that subject is a collaborator on
+// the namespace to which the CollaboratorMatcher belongs.  If any one of those
+// CollaboratorMatcher objects matches a subject, that subject is considered to
+// be a collaborator on the namespace.
+type CollaboratorMatcher struct {
+	unversioned.TypeMeta
+
+	// Standard object's metadata.
+	kapi.ObjectMeta
+
+	// Spec defines the matcher.
+	Spec CollaboratorMatcherSpec
+}
+
+// CollaboratorMatcherSpec defines the collaborator matcher.  Exactly one field
+// must be non-nil.
+type CollaboratorMatcherSpec struct {
+	// UserMatcher matches against user subjects.
+	UserMatcher *UserMatcher
+
+	// GroupMatcher matches against group subjects.
+	GroupMatcher *GroupMatcher
+
+	// ServiceAccountMatcher matches against service-account subjects.
+	ServiceAccountMatcher *ServiceAccountMatcher
+}
+
+// CollaboratorMatcherList is a collection of CollaboratorMatcher objects.
+type CollaboratorMatcherList struct {
+	unversioned.TypeMeta
+
+	// Standard object's metadata.
+	unversioned.ListMeta
+
+	// Items is a list of CollaboratorMatcher objects.
+	Items []CollaboratorMatcher
+}
+
+// UserMatcher matches a user either by a string match on the user name,
+// a string name on the name of a group to which the user belongs, or
+// a label selector.
+type UserMatcher struct {
+	// Users specifies a list of literal user names.
+	Users []string
+
+	// Groups specifies a list of literal group names.
```

Comment:
`Groups is a list of groups used to match against an individual user's groups.  If the user is a member of one of the whitelisted groups, he's allowed to be bound to a role.`